### ISSUE TITLE
run emptydrops after 10x cell ranger 3.0

### ISSUE THREAD

**Issue by zji90** [MarioniLab/DropletUtils#17]
Hi,
I wonder whether it is necessary to run emptydrops function for count matrix obtained from 10x cell ranger 3.0 and above? The following description for cell ranger indicates that the algorithm is already using emptydrops internally?

'Cell Ranger 3.0 introduces an improved cell-calling algorithm that is better able to identify populations of low RNA content cells, especially when low RNA content cells are mixed into a population of high RNA content cells. For example, tumor samples often contain large tumor cells mixed with smaller tumor infiltrating lymphocytes (TIL) and researchers may be particularly interested in the TIL population. The new algorithm is based on the EmptyDrops method (Lun et al., 2018).'

**Comment by LTLA** [comment]
Yes, version 3 uses their own implementation of `EmptyDrops`. I haven't poked inside but I assume that they implemented it faithfully. However, IIRC, their algorithm is slightly different with respect to how they set `retain` - I think they use their old threshold, so as to guarantee that version 3 detects a superset of barcode detected by version 2. And of course, the Python PRNG has different settings from the R PRNG, so even if they implemented it perfectly, you'd probably still get different numbers.

**tl;dr** for routine applications, you don't have to run `emptyDrops()` separately unless you want to have more control or better diagnostics.

**Comment by zji90** [comment]
Thanks for your advice!
